src/corefx/System.Globalization.Native/CMakeLists.txt

```diff
@@ -68,6 +71,12 @@ _add_library(System.Globalization.Native_Static
 # Disable the "lib" prefix.
 set_target_properties(System.Globalization.Native PROPERTIES PREFIX "")
 
+# Force CMake to not link against libstdc++
+# http://cmake.3232098.n2.nabble.com/setting-LINKER-LANGUAGE-still-adds-lstdc-td7581940.html
```

Comment:
Can you try it? This is the only option that works.

Reply:
I've tried to remove the changes here and it has built fine on my machine. Maybe you didn't do clean build and something has leaked from the previous state? Just try to deleted the whole bin folder before the build.

Reply:
I've used cmake 3.5.1 and clang 3.9.1

Reply:
This wasn't about the build failure, if you follow the comments above (https://github.com/dotnet/coreclr/pull/18614#issuecomment-399690629), it was found with `ldd System.Globalization.Native.so` that we are still linking against lddstdc++. With this change it fixed that issue.

Reply:
Ah, I am sorry, I've missed that. But I've found the real culprit. The problem is that we are including one .cpp file in the build of the library - the version.cpp that is compiled into all targets to provide info on the version of the library. This is automatically added to all targets, so that's why you have not noticed it. See the `_add_library` function in `functions.cmake`.
Modifying the generator of version.cpp to generate file named version.c instead and updating the VERSION_FILE_PATH variable in the root CMakeLists.txt accordingly fixes the problem..

Reply:
Nice, thanks. I have renamed version.cpp to version.c and removed these hacks. Checked with `ldd` no libstdc++.